tests/IntegrationTest/BaseLineTest.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/**
+ * This file is part of the Dealerdirect PHP_CodeSniffer Standards
+ * Composer Installer Plugin package.
+ *
+ * @copyright 2022 PHPCodeSniffer Composer Installer Contributors
+ * @license MIT
+ */
+
+namespace Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests\IntegrationTest;
+
+use Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests\PHPCSVersions;
+use Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer\Tests\TestCase;
+
+/**
+ * Test baseline.
+ *
+ * These tests verify:
+ * - That the plugin can be installed and functions correctly with the full range of supported PHPCS versions.
+ *   While the test is not run against the full range of supported PHPCS version each time, due to the
+ *   random PHPCS version selection, all supported PHPCS versions will be tested over time in CI.
+ * - That the plugin runs when installed.
+ * - That the PHPCS native standards are the only recognized standards when no external standards are available.
+ * - That no `CodeSniffer.conf``file gets created when no external standards are found.
+ */
+final class BaseLineTest extends TestCase
+{
+    private $composerConfig = array(
+        'name'        => 'phpcs-composer-installer/baseline-test',
+        'require-dev' => array(
+            'squizlabs/php_codesniffer'                      => null,
+            'dealerdirect/phpcodesniffer-composer-installer' => '*',
+        ),
+    );
+
+    /**
+     * Set up test environment before each test.
+     */
+    protected function set_up()
+    {
+        $this->createTestEnvironment();
+    }
+
+    /**
+     * Clean up after each test.
+     */
+    protected function tear_down()
+    {
+        $this->removeTestEnvironment();
+    }
+
+    /**
+     * Baseline test for a Composer GLOBAL install.
+     *
+     * @dataProvider dataBaseLine
+     *
+     * @param string $phpcsVersion  PHPCS version to use in this test.
+     *                              This version is randomly selected from the PHPCS versions compatible
+     *                              with the PHP version used in the test.
+     * @param array  $expectedStnds List of the standards which are expected to be registered.
+     *
+     * @return void
+     */
+    public function testBaseLineGlobal($phpcsVersion, $expectedStnds)
+    {
+        $config                                             = $this->composerConfig;
+        $config['require-dev']['squizlabs/php_codesniffer'] = $phpcsVersion;
+
+        $this->writeComposerJsonFile($config, static::$tempGlobalPath);
```

Comment:
As the BaseLineTest class is `final` (and we don't expect the class to be extended anyway), `self` can be used instead of `static`.

Reply:
True, it's more of a consistency thing to always use `static::` unless I explicitly want `self` or `parent`.

I have no objections to changing it though (though will need to change it everywhere as I have used it everywhere).

Reply:
Its fine to leave as-is. Its just something that stood out when reviewing.